sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -111,6 +112,39 @@ object RapidsPluginUtils extends Logging {
     }
   }
 
+  private def detectMultipleJar(propNames: Seq[String], jarName: String, conf: RapidsConf): Unit = {
+    val classloader = ShimLoader.getShimClassLoader()
+    val rapidsJarURLs = propNames.flatMap(propName =>
+      classloader.getResources(propName).asScala
+    ).toSet
+    lazy val rapidsJars = rapidsJarURLs.map(_.toString.split("!").head).mkString(",")
+    lazy val rapidsJarsVers = rapidsJarURLs.map { 
+      url => scala.io.Source.fromInputStream(url.openStream()).mkString("") 
+    }.mkString(",")
+    lazy val msg = s"Multiple $jarName jars found in the classpath: $rapidsJars, please make " + 
+        s"sure there is only one $jarName jar in the classpath. If it is impossible to fix the " +
+        s"classpath you can suppress the error by setting ${RapidsConf.ALLOW_MULTIPLE_JARS.key}" + 
+        s"to true. Version info: \n$rapidsJarsVers"
+    conf.allowMultipleJars match {
+      case true => {
+        if (rapidsJarURLs.size > 1) {
```

Comment:
Now we have packed the cuDF and JNI jar into the Rapids plugin jar, is it a multiple jars case when there are a Rapids jar and JNI jar in the classpath (or a Rapids jar and a cuDF jar)?
How do we define this multiple jars case ?

Reply:
> Now we have packed the cuDF and JNI jar into the Rapids plugin jar, is it a multiple jars case when there are a Rapids jar and JNI jar in the classpath (or a Rapids jar and a cuDF jar)?

Yes it is.

> How do we define this multiple jars case ?

I think it means the cases where it is not clear to the plugin which jar to load.

If there is a Rapids jar and a cuDF jar, we always want to load the cuDF jar inside the plugin jar and avoid loading the cuDF jar because it might be an old jar.

Reply:
Then seems we should not call the `detectMultipleJar` with cudf jar and JNI jar separately.
Instead something like
`detectMultipleJar(Seq(propsName212, propsName213, propsNameOld) ++ Seq(JNI_PROPS_FILENAME, CUDF_PROPS_FILENAME))`.

And we'd better document this case in the warning message.